src/lib/template.ts

```diff
@@ -104,61 +99,49 @@ export class Template {
+            partIndex += lastIndex;
+            // Generate a new text node for each literal section
+            // These nodes are also used as the markers for node parts
+            for (let i = 0; i < lastIndex; i++) {
+              parent.insertBefore(
+                  (strings[i] === '') ? createMarker() :
+                                        document.createTextNode(strings[i]),
+                  node);
+              this.parts.push({type: 'node', index: ++index});
+            }
             parent.insertBefore(
-                (strings[i] === '') ? createMarker() :
-                                      document.createTextNode(strings[i]),
+                strings[lastIndex] === '' ?
+                    createMarker() :
+                    document.createTextNode(strings[lastIndex]),
                 node);
-            this.parts.push({type: 'node', index: index++});
+            nodesToRemove.push(node);
           }
-          parent.insertBefore(
-              strings[lastIndex] === '' ?
-                  createMarker() :
-                  document.createTextNode(strings[lastIndex]),
-              node);
-          nodesToRemove.push(node);
         } else if (node.nodeType === 8 /* Node.COMMENT_NODE */) {
           if (node.nodeValue === marker) {
             const parent = node.parentNode!;
             // Add a new marker node to be the startNode of the Part if any of
             // the following are true:
             //  * We don't have a previousSibling
-            //  * previousSibling is being removed (thus it's not the
-            //    `previousNode`)
-            //  * previousSibling is not a Text node
-            //
-            // TODO(justinfagnani): We should be able to use the previousNode
-            // here as the marker node and reduce the number of extra nodes we
-            // add to a template. See
-            // https://github.com/PolymerLabs/lit-html/issues/147
-            const previousSibling = node.previousSibling;
-            if (previousSibling === null || previousSibling !== previousNode ||
-                previousSibling.nodeType !== Node.TEXT_NODE) {
+            //  * The previousSibling is already the start of a previous part
+            if (node.previousSibling === null || index === lastStartIndex) {
+              index++;
               parent.insertBefore(createMarker(), node);
-            } else {
-              index--;
             }
-            this.parts.push({type: 'node', index: index++});
-            nodesToRemove.push(node);
-            // If we don't have a nextSibling add a marker node.
-            // We don't have to check if the next node is going to be removed,
-            // because that node will induce a new marker if so.
+            lastStartIndex = index;
+            this.parts.push({type: 'node', index});
+            // If we don't have a nextSibling, keep this node.
+            // Else, remove it to save memory.
```

Comment:
We remove the node to reduce clone costs as well, so maybe generalize the reason.